bundles/org.openhab.extensionservice.marketplace/src/main/java/org/openhab/extensionservice/marketplace/internal/MarketplaceProxy.java

```diff
@@ -72,7 +71,7 @@ public MarketplaceProxy() {
      * @return list of marketplace nodes
      */
     public List<Node> getNodes() {
-        return cachedNodes != null ? Arrays.asList(cachedNodes) : Collections.emptyList();
+        return Arrays.asList(cachedNodes);
```

Comment:
But it makes sense to return an immutable list here. Doesn't it?

```suggestion
        return Collections.unmodifiableList(Arrays.asList(cachedNodes));
```

Reply:
Yes it's best to prevent code from messing with the cached values.